sdk-extensions/autoconfigure/src/test/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java

```diff
@@ -367,4 +370,15 @@ void tracerProviderCustomizer() {
     // Ensures the export happened.
     sdk.getSdkTracerProvider().shutdown().join(10, TimeUnit.SECONDS);
   }
+
+  @Test
+  void testNonStringProperties() {
+    Properties properties = System.getProperties();
+
+    properties.putIfAbsent("my-key", 7);
+    properties.putIfAbsent(7.39, "my-value");
+    properties.putIfAbsent(new BigDecimal("7.39"), new BigInteger("7"));
+
+    builder.build();
```

Comment:
Thanks - let's also go ahead and access the properties, using the string coerced form.

Reply:
How can I access to the `ConfigProperty` instance from the `AutoConfiguredOpenTelemetrySdkBuilder`? I think that method is private.

Reply:
never mind I used AssertJ to extract the field